File: API/Entities/Product.cs
using System.Collections.Generic;

namespace API.Entities
{
    public class Product
    {
        public int Id { get; set; }

        public string Name { get; set; }
        public string Vendor { get; set; }
        public string Description { get; set; }

        public int CategoryId { get; set; }
        public Category Category { get; set; }

        public ICollection<ProductImg> ProductImgs { get; set; }
        public ICollection<ProductFeature> ProductFeatures { get; set; }
        public ICollection<RealProduct> RealProducts { get; set; }
    }
}

File: client/src/app/_models/product.ts
import { ProductFeature } from "./productFeature";
import { RealProduct } from "./realProduct";

export interface Product {
    id: number;
    name: string;
    vendor: string;
    description: string;
    categoryId: number;
    categoryName: string;
    productImgsLinks: string[];
    productFeatures: ProductFeature[];
    realProducts: RealProduct[];
}